Lesson4.ipynb

```diff
@@ -29,6 +32,655 @@
+    "    header = readCSV.fieldnames\n",
+    "    NamesForStepsMoreThan100K = ''\n",
+    "    \n",
+    "    for row in readCSV:\n",
+    "        totalSteps = sumStepsPerPerson(row, header)\n",
+    "        if totalSteps >= 100000:\n",
+    "            NamesForStepsMoreThan100K = NamesForStepsMoreThan100K + row['team_member'] + ' '\n",
+    "    print(\"Steps more than 100K:\" + NamesForStepsMoreThan100K)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Follow the code block above, can you complete a code block to print the names of people with steps count more than 300k and 400k**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Write you code here:\n",
+    "\n",
+    "# Read from CSV file\n",
+    "# Count for total steps\n",
+    "# Compare with 300K/400k\n",
+    "# Print the names\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Print total steps for each team\n",
+    "\n",
+    "From the code above, we complete a function for counting total steps for each attendee.\n",
+    "\n",
+    "And what else we should do? -- We should know **all the team names**.\n",
+    "\n",
+    "Let's write more functions.  \n",
+    "\n",
+    "### Get all the team names\n",
+    "\n",
+    "To store the team names, create a list.\n",
+    "\n",
+    "Please refer to the code block below:\n",
+    "\n",
+    "```python\n",
+    "def getTeamNamesFromCSV():\n",
+    "    ...\n",
+    "    ...\n",
+    "    team_names = []                              # Create a list to store all the team names\n",
+    "    for row in readCSV:                          # Loop the CSV file to get the team name from each row\n",
+    "        if row['team_name'] not in team_names:   # If the team name is not in the list, then append it to the list\n",
+    "            team_names.append(row['team_name'])\n",
+    "    return team_names\n",
+    "```\n",
+    "** **\n",
+    "### Count total steps by team name \n",
+    "\n",
+    "After getting the team name, the next step is count the total steps by team name.\n",
+    "\n",
+    "Please refer to the code block below:\n",
+    "\n",
+    "```python\n",
+    "def getTotalStepsByTeamNameFromCSV(team_name):\n",
+    "    ...\n",
+    "    ...\n",
+    "    teamTotalSteps = 0\n",
+    "    header = readCSV.fieldnames\n",
+    "    for row in readCSV:\n",
+    "        if row['team_name'] == team_name:        # If the \"team_name\" in a row equals the target team_name, then plus it\n",
+    "            teamTotalSteps = teamTotalSteps + totalStepsByRow(row, header)\n",
+    "```\n",
+    "** **\n",
+    "### Loop for team names\n",
+    "\n",
+    "To print the total steps for each team, let's do a loop for the team names outside the code block above.\n",
+    "\n",
+    "Please refer the code block below:\n",
```

Comment:
"Please refer the code block below" should read ""Please refer to the code block below"